Tools/ardupilotwaf/chibios.py

```diff
@@ -624,17 +628,27 @@ def generate_hwdef_h(env):
         print(env.BOOTLOADER_OPTION)
         env.BOOTLOADER_OPTION += " --signed-fw"
         print(env.BOOTLOADER_OPTION)
-    hwdef_script = os.path.join(env.SRCROOT, 'libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py')
     hwdef_out = env.BUILDROOT
     if not os.path.exists(hwdef_out):
         os.mkdir(hwdef_out)
-    python = sys.executable
-    cmd = "{0} '{1}' -D '{2}' --params '{3}' '{4}'".format(python, hwdef_script, hwdef_out, env.DEFAULT_PARAMETERS, env.HWDEF)
+
+    # Build a list of the hardware definitions
+    hwdef_arr = [env.HWDEF]
     if env.HWDEF_EXTRA:
-        cmd += " '{0}'".format(env.HWDEF_EXTRA)
-    if env.BOOTLOADER_OPTION:
-        cmd += " " + env.BOOTLOADER_OPTION
-    return subprocess.call(cmd, shell=True)
+        # Add any extra hwdefs that may be defined
+        hwdef_arr.append(env.HWDEF_EXTRA)
+    ch = chibios_hwdef.ChibiOSHWDef(outdir=hwdef_out,
+                                    bootloader=env.BOOTLOADER,
+                                    signed_fw=env.AP_SIGNED_FIRMWARE,
```

Comment:
`ChibiOSHWDef` defines `signed_fw` as bool.

If waf is called with `--signed-fw` then `env.AP_SIGNED_FIRMWARE` is set to `True`.
If waf is called without `signed-fw` then `env.AP_SIGNED_FIRMWARE` is set to `[ ]` (empty list)

Given python considers empty list as False, this might be safe. The last commit changes `wscript` treat `AP_SIGNED_FIRMWARE` similarly to `BOOTLOADER`, setting the env flag based on what is passed to it at build time.